### PR TITLE
Base task for terratorch 

### PR DESCRIPTION
**Pull request by Joao-L-S-Almeida**
By creating a parent class for all the tasks we avoid code duplication, as aimed by issue #153. In the same way, it is easier to modify and fix bugs for common methods.

### CODE REVIEW

**Review by romeokienzler — APPROVED** [review]
@Joao-L-S-Almeida thanks a lot, lgtm

- [x] merged

File: terratorch/tasks/segmentation_tasks.py [hunk 1]
```diff
@@ -275,79 +237,6 @@ def training_step(self, batch: Any, batch_idx: int, dataloader_idx: int = 0) ->
...
-        return (
-            batch_index < BATCH_IDX_FOR_VALIDATION_PLOTTING
-            and hasattr(self.trainer, "datamodule")
-            and self.logger
-            and not self.current_epoch % self.plot_on_val  # will be True every self.plot_on_val epochs
-            and hasattr(self.logger, "experiment")
-            and (hasattr(self.logger.experiment, "add_figure") or hasattr(self.logger.experiment, "log_figure"))
-        )
-
-    def validation_step(self, batch: Any, batch_idx: int, dataloader_idx: int = 0) -> None:
```
[the remark this answers is not in the document]
Comment (Joao-L-S-Almeida) [reply terratorch/tasks/segmentation_tasks.py:296]: Hi. Wrong step. Putting it back. Thanks.

[the remark this answers is not in the document]
Comment (blumenstiel) [reply terratorch/tasks/segmentation_tasks.py:296]: Thanks!

[the remark this answers is not in the document]
Comment (Joao-L-S-Almeida) [reply terratorch/tasks/segmentation_tasks.py:296]: I sent #311 to fix it.

**Pull request merged**